java/arcs/android/storage/database/DatabaseImpl.kt

```diff
@@ -120,15 +120,24 @@ class DatabaseImpl(
         readableDatabase.useTransaction {
             val db = this
             // Fetch the entity's type by storage key.
-            val (storageKeyId, schemaTypeId) = rawQuery(
-                "SELECT id, value_id, data_type FROM storage_keys WHERE storage_key = ?",
+            val (storageKeyId, schemaTypeId, entityId) = rawQuery(
+                """
+                    SELECT
+                        storage_keys.id,
+                        storage_keys.value_id,
+                        storage_keys.data_type,
+                        entities.entity_id
+                    FROM storage_keys
+                    LEFT JOIN entities ON storage_keys.id = entities.storage_key_id
```

Comment:
A `LEFT JOIN` here will result in a null `entities.entity_id` column in the results if there is no entity record for the given storage key. I think this isn't quite what we want - we'd prefer that no records are returned in this scenario, right?

Reply:
It was for the error reporting. If you try to get an Entity, but the stored data is, say, a Collection, then a `JOIN` will return no results and it will throw "Entity at storage key $storageKey does not exist".

If you do the `INNER JOIN` then you'll get a result, but with a null `entity_id`. Then instead it will throw the more helpful error message "Expected storage key $storageKey to be an Entity but was a $dataType."

Reply:
Gotcha, that makes sense to me 👍